paf-mvp-core-js/src/ui/fields.ts

```diff
@@ -99,24 +158,62 @@ export abstract class Field<T, M extends IModel> extends FieldReadOnly<T, M> imp
   }
 
   /**
-   * Sets the current value, updating any HTML elements that match the name value, then checks to see if any other
-   * fields need to be updated if the model is not already in a setting values operation.
+   * Sets the current value, then checks to see if any other fields need to be updated if the model is not already in a
+   * setting values operation. Finally updates any UI elements bound to the model.
+   * @remarks if the field is disabled an exception is thrown.
    */
   public set value(value: T) {
+    if (this._disabled) {
+      throw "Can't set value of disabled field";
+    }
     this._value = value;
-    this.bindings.forEach((b) => b.setValue(value));
     if (this.model.settingValues === false) {
       this.model.settingValues = true;
       this.updateOthers();
       this.model.settingValues = false;
     }
+    this.model.updateUI();
+  }
+
+  /**
+   * Gets the value that was persisted, or undefined if there is no persisted value.
+   */
+  public get persistedValue(): T {
+    return this._persistedValue;
+  }
+
+  /**
+   * Sets the current value and the persisted value, then checks to see if any other fields need to be updated if the
+   * model is not already in a setting values operation. Finally updates any UI elements bound to the model.
+   * Should be used instead of the value property when setting a value from persisted storage.
+   */
+  public set persistedValue(value: T) {
+    this._persistedValue = value;
+    this.value = value;
+  }
+
+  /**
+   * True if the field is disabled and can't have it's value changed. Should be overridden in inheriting classes if the
+   * ability to disable a field is needed.
+   */
+  public get disabled(): boolean {
+    return this._disabled;
+  }
+
+  /**
+   * Ensures the UI components are updated if the disabled status of the field changes.
+   */
+  public set disabled(value: boolean) {
+    this._disabled = value;
+    this.bindings.forEach((b) => b.refresh());
   }
 
   /**
-   * Resets the field to the original value.
+   * Resets the field and persisted value to the default value.
    */
   reset() {
     this.value = this.defaultValue;
+    this.persistedValue = undefined;
```

Comment:
⚠️ this will also set the `value` to `undefined` (line 192), which makes the previous line useless.